tests/src/main/java/com/hazelcast/stabilizer/tests/map/MapHeapHogWarmupTest.java

```diff
@@ -0,0 +1,136 @@
+import static com.hazelcast.stabilizer.tests.helpers.HazelcastTestUtils.nextKeyOwnedBy;
+import static com.hazelcast.stabilizer.utils.CommonUtils.sleepSeconds;
+import static java.lang.String.format;
+
+/**
+ * Fills up the cluster to a defined heap usage factor during warmup phase.
+ * <p/>
+ * This tests intentionally has an empty run phase.
+ */
+public class MapHeapHogWarmupTest {
+
+    private static final int MIN_CLUSTER_SIZE = 3;
+    private static final long APPROX_ENTRY_BYTES_SIZE = 238;
+    private static final ILogger LOGGER = Logger.getLogger(MapHeapHogWarmupTest.class);
+
+    // properties
+    public String basename = this.getClass().getSimpleName();
+    public int threadCount = 10;
+    public int ttlHours = 24;
+    public double approxHeapUsageFactor = 0.9;
+
+    private HazelcastInstance targetInstance;
+    private ThreadSpawner threadSpawner;
+    private IMap<Long, Long> map;
+
+    private long maxEntriesPerThread;
+
+    @Setup
+    public void setup(TestContext testContext) {
+        targetInstance = testContext.getTargetInstance();
+        threadSpawner = new ThreadSpawner(testContext.getTestId());
+        map = targetInstance.getMap(basename);
+
+        // calculate how many entries we have to insert per member and thread
+        long free = Runtime.getRuntime().freeMemory();
+        long total = Runtime.getRuntime().totalMemory();
+        long max = Runtime.getRuntime().maxMemory();
+
+        long used = total - free;
+        long totalFree = max - used;
+
+        long maxLocalEntries = (long) ((totalFree / APPROX_ENTRY_BYTES_SIZE) * approxHeapUsageFactor);
+        maxEntriesPerThread = (long) ((double) maxLocalEntries / (double) threadCount);
+    }
+
+    @Warmup()
+    public void warmup() {
+        if (isClient(targetInstance)) {
+            return;
+        }
+
+        // wait for the cluster to build up
+        while (targetInstance.getCluster().getMembers().size() != MIN_CLUSTER_SIZE) {
+            LOGGER.info(basename + " waiting to hit cluster size " + MIN_CLUSTER_SIZE);
+            sleepSeconds(1);
+        }
+
+        // fill the cluster as fast as possible with data
+        long startKey = 0;
+        for (int i = 0; i < threadCount; i++) {
+            threadSpawner.spawn(new FillMapWorker(startKey));
+            startKey += maxEntriesPerThread;
+        }
+        threadSpawner.awaitCompletion();
+
+        StringBuilder sb = new StringBuilder("After warmup of ").append(basename).append("\nmap size = ").append(map.size());
+        addMemoryStatistics(sb);
+        LOGGER.info(sb.toString());
+    }
+
+    @Verify()
+    public void verify() {
+        if (isClient(targetInstance)) {
+            return;
+        }
+
+        StringBuilder sb = new StringBuilder("Local verify of ").append(basename).append("\nmap size = ").append(map.size());
+        addMemoryStatistics(sb);
+        LOGGER.info(sb.toString());
+    }
+
+    @Run
+    public void run() {
```

Comment:
Yes I think that is the best idea.  

we have, 3 types of "test"

1  Validation Test  the normal type of test, where we are trying to, catch a memory leak, or stress our system,  through some actions.   in this type of test we don't really care about the throughput or latency of some odd set of operations,  just if it did not crash and we get the result we wanted.

2, performance test,  where all we care is throughput and latency of some set of operations.

3,  system/setup "test",  like this one where the "test" is only really used to create a cluster in some state,  and the "test" part of it  is carried out  in a different way,  say by blocking the ip tables of a member or terminating the box a member is running on,  or creating a higher network load with iperf tool running against one box,  ect ect ect.